web/ui/static/js/graph/index.js

```diff
@@ -241,48 +251,55 @@ Prometheus.Graph.prototype.populateInsertableMetrics = function() {
           map: {}
         }
 
-        self.expr.typeahead({
-          source: metrics,
-          items: "all",
-          matcher: function(item) {
-            // If we have result for current query, skip
-            if (self.fuzzyResult.query !== this.query) {
-              self.fuzzyResult.query = this.query;
-              self.fuzzyResult.map = {};
-              self.fuzzyResult.result = fuzzy.filter(this.query.replace(/ /g, ''), metrics, {
-                pre: '<strong>',
-                post: '</strong>'
-              });
-              self.fuzzyResult.result.forEach(function(r) {
-                self.fuzzyResult.map[r.original] = r;
-              });
-            }
-
-            return item in self.fuzzyResult.map;
-          },
-
-          sorter: function(items) {
-            items.sort(function(a,b) {
-              var i = self.fuzzyResult.map[b].score - self.fuzzyResult.map[a].score;
-              return i === 0 ? a.localeCompare(b) : i;
-            });
-            return items;
-          },
-
-          highlighter: function (item) {
-            return $('<div>' + self.fuzzyResult.map[item].string + '</div>')
-          },
-        });
-        // This needs to happen after attaching the typeahead plugin, as it
-        // otherwise breaks the typeahead functionality.
-        self.expr.focus();
+        self.initTypeahead(self);
       },
       error: function() {
         self.showError("Error loading available metrics!");
       },
   });
 };
 
+Prometheus.Graph.prototype.initTypeahead = function(self) {
+  const historyIsChecked = $("div.query-history button").hasClass("is-checked");
+  const source = historyIsChecked ? JSON.parse(localStorage.getItem("history")) : self.allMetrics;
+
+  self.expr.typeahead({
+    source,
+    items: "all",
+    matcher: (item) => {
```

Comment:
So, the consensus is that arrow functions are not widely supported (On mobile, primarily) and its not worth breaking anyone for a couple of lines of code.